modules/REST/V1/Settings.php

```diff
@@ -106,10 +114,19 @@ public function update_item( $request ) {
 		);
 
 		foreach ( $options as $option_name => $new_value ) {
-			$result = $this->options->set( $option_name, $new_value );
+			$prev_value = $this->options->get( $option_name );
+			$result     = $this->options->set( $option_name, $new_value );
 
 			if ( $result->has_errors() ) {
 				$errors->merge_from( $result );
+				continue;
+			}
+			if ( $prev_value === $new_value ) {
+				continue;
+			}
+			// Default language.
```

Comment:
```suggestion
```
I think the code speaks for itself.

Reply:
Am I not even allowed to manspain my own code?!
Done in [6af6129](https://github.com/polylang/polylang/pull/1594/commits/6af61290e152f4bb776b6b0087b4267436e24bd9) 😬